src/core/algorithms/od/fastod/single_attribute_predicate.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <string>
+
+#include "data_frame.h"
+
+namespace algos::fastod {
+
+struct SingleAttributePredicate {
```

Comment:
Can you explain where we use this class?

Reply:
Previously we used it in partition classes. But at some point we abandoned it. Removed it.